.github/workflows/build-and-deploy.yml

```diff
@@ -29,6 +29,20 @@ jobs:
       TF_VAR_version_identifier: ${{ github.sha }}
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/cache@v3
```

Comment:
We discussed this a bit earlier today, but circling back here for posterity:
1. Yes, that's correct. However, it looks like [actions/setup-go](https://github.com/actions/setup-go)'s default caching behavior is working just fine to handle the go cache, so I plan on removing the second usage of `actions/cache@v3` here.
    - At any rate, I'm not sure it's useful to have separate cache save/restore actions in the same workflow, so _if_ we were going to keep using that for the go cache, it would probably be faster to combine both usages.
2. Yeah, I think `Linux-taskfile` is probably sufficient for this. Given that everything making use of this cache will independently verify its state, I think there's low/no risk in restoring a stale cache. Speaking only for the usages in this repository, including a file hash in the cache key is really more about preempting a useless cache restore that would just be discarded/overwritten later in the job – i.e. it's more about saying "don't bother, I already know the cache is busted" (e.g. because `go.sum` has been modified or something) than avoiding _problems_ arising from a stale cache getting restored.
3. Good call, will add.